reward/staking_manager.go

```diff
@@ -242,6 +246,101 @@ func GetStakingInfoOnStakingBlock(stakingBlockNumber uint64) *StakingInfo {
 	return calcStakingInfo
 }
 
+// PreloadStakingInfo preloads staking info for the given headers.
+// It first finds the first block that does not have state, and then
+// it regenerates the state from the nearest block that has state to the target block to preload staking info.
+// Note that the state is saved every 128 blocks to disk in full node.
+func PreloadStakingInfo(headers []*types.Header) ([]uint64, error) {
+	// If no headers to preload, do nothing
+	if len(headers) == 0 {
+		return nil, nil
+	}
+
+	if stakingManager == nil {
+		return nil, ErrStakingManagerNotSet
+	}
+
+	var (
+		current  *types.Block
+		database state.Database
+		target   = headers[len(headers)-1].Number.Uint64()
+		bc       = stakingManager.blockchain
+	)
+
+	database = state.NewDatabaseWithExistingCache(bc.StateCache().TrieDB().DiskDB(), bc.StateCache().TrieDB().TrieNodeCache())
+
+	// Find the first block that does not have state
+	i := 0
+	for i < len(headers) {
+		if _, err := state.New(headers[i].Root, database, nil, nil); err != nil {
+			break
+		}
+		i++
+	}
+	// Early return if all blocks have state
+	if i == len(headers) {
+		return nil, nil
+	}
+
+	// Find the nearest block that has state
+	origin := headers[i].Number.Uint64() - headers[i].Number.Uint64()%128
```

Comment:
How about introducing a getter of `bc.cacheConfig.BlockInterval` and using it?

Reply:
Better way I think. Add it later. Thanks.